app/app.go

```diff
@@ -328,6 +336,43 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
 	// If evidence needs to be handled for the app, set routes in router here and seal
 	app.EvidenceKeeper = *evidenceKeeper
 
+	// just re-use the full router - do we want to limit this more?
+	var wasmRouter = bApp.Router()
+	wasmDir := filepath.Join(homePath, "wasm")
+
+	wasmWrap := WasmWrapper{Wasm: wasm.DefaultWasmConfig()}
```

Comment:
Out of curiosity, did you follow some tutorial/docs from CosmWasm for this?

Reply:
I think he's following the docs/instructions from cosmwasm/wasmd integration

Reply:
Why do we need this wrapper?

Reply:
to create wasm keeper we need to pass wasm config which is using this wrapper.